src/main/java/com/aws/greengrass/ipc/SubscribeToCertificateUpdatesOperationHandler.java

```diff
@@ -0,0 +1,190 @@
+    private static final int RSA_KEY_LENGTH_FOR_SERVER_CERT = 4096;
+    private final String serviceName;
+    private final ClientDevicesAuthService clientDevicesAuthService;
+    private final CertificateManager certificateManager;
+    private final AtomicBoolean subscriptionResponseSent = new AtomicBoolean(false);
+    private final AtomicReference<CertificateUpdateEvent> firstStreamingEvent = new AtomicReference<>(null);
+    private KeyPair keyPair;
+
+    /**
+     * Constructor.
+     *
+     * @param context                  operation continuation handler
+     * @param clientDevicesAuthService client device auth service
+     */
+    public SubscribeToCertificateUpdatesOperationHandler(OperationContinuationHandlerContext context,
+                                                         ClientDevicesAuthService clientDevicesAuthService) {
+        super(context);
+        serviceName = context.getAuthenticationData().getIdentityLabel();
+        this.clientDevicesAuthService = clientDevicesAuthService;
+        this.certificateManager = clientDevicesAuthService.getCertificateManager();
+
+    }
+
+    @Override
+    protected void onStreamClosed() {
+        //TODO: Unsubscribe to cert updates.
+    }
+
+    /**
+     * Override to appropriately enforce stream responses are sent after the initial response.
+     * This only gets called if handleRequest returns normally and starts to send a response.
+     * <p>
+     * Here, we know that the initial SubscribeToCertificateUpdatesResponse response is sent. So, we send the
+     * first streaming response of CertificateUpdateEvent.
+     * </p>
+     */
+    @Override
+    public void afterHandleRequest() {
+        synchronized (firstStreamingEvent) {
+            if (subscriptionResponseSent.compareAndSet(false, true)) {
+                CertificateUpdateEvent event = firstStreamingEvent.get();
+                if (event != null) {
+                    sendStreamEvent(event);
+                }
+                firstStreamingEvent.set(null);
+            }
+        }
+    }
+
+    @SuppressWarnings("PMD.PreserveStackTrace")
+    @Override
+    public SubscribeToCertificateUpdatesResponse handleRequest(
+            SubscribeToCertificateUpdatesRequest subscribeToCertificateUpdatesRequest) {
+        return translateExceptions(() -> {
+            try {
+                clientDevicesAuthService.doAuthorizationForSubscribingToCertUpdates(serviceName);
+            } catch (AuthorizationException e) {
+                logger.atInfo().kv("error", e.getMessage()).kv(COMPONENT_NAME, serviceName).log(UNAUTHORIZED_ERROR);
+                throw new UnauthorizedError(e.getMessage());
+            }
+            try {
+                CertificateType certificateType = getCertificateTypeFromCertificateOptions(
+                        subscribeToCertificateUpdatesRequest.getCertificateOptions());
+                if (CertificateType.SERVER.equals(certificateType)) {
```

Comment:
Log the `else` case. Unknown certificate type. This could help future customers if we add support for client certs and they're running a version of CDA that's too old

...unless `certificateOptions.getCertificateType()` returns NULL on an unknown type? I suppose it does

Reply:
If the certificate type is invalid, it is considered as null and `InvalidArgumentsError` is thrown. I updated the error message and also added a test for the same.